src/TextField/TextField.js

```diff
@@ -73,14 +72,14 @@ const getStyles = (props, context, state) => {
     font: 'inherit',
   });
 
-  if (state.isFocused) {
-    styles.floatingLabel.color = focusColor;
-  }
-
   if (state.hasValue) {
     styles.floatingLabel.color = fade(props.disabled ? disabledTextColor : floatingLabelColor, 0.5);
   }
 
+  if (state.isFocused) {
+    styles.floatingLabel.color = focusColor;
+  }
```

Comment:
@nathanmarks Switching the order here fixes the focus color issue. Unless I'm missing why hasValue was overriding it.